### PR TITLE
Add reproject docs

### PR DESCRIPTION
**Pull request by keflavich**
Solves #590.

It might be helpful to add a little more documentation, but the basics are here and the reference to [reproject](https://reproject.readthedocs.io/en/stable/) is perhaps enough to get people going.

### CODE REVIEW

File: docs/smoothing.rst [hunk 1]
```diff
@@ -160,3 +160,27 @@ done in the unit of pixel.  In our example, each channel is 0.1 km/s wide::
...
 By using `gaussian_width.value`, we convert the pixel width into a float.
+
+Reprojection
+============
+
+Smoothing changes the data properties but not the underlying grid.  It is often
+helpful to re-project the data in either the spatial or spectral directions to
+match cubes in pixel space.
+
+Spatial reprojection can be achieved with the `SpectralCube.reproject` method::
```
Comment (e-koch) [inline docs/smoothing.rst:171]:
```suggestion
Spatial reprojection can be achieved with the :class:`~spectral_cube.SpectralCube.reproject` method::
```
Unsure if this is still needed to generate the link.

File: docs/smoothing.rst [hunk 2]
```diff
@@ -160,3 +160,27 @@ done in the unit of pixel.  In our example, each channel is 0.1 km/s wide::
...
+
+    reproj_cube = cube.reproject(new_header)
+
+This method will loop over each channel using the `reproject
+<https://reproject.readthedocs.io/en/stable/>`_ model to regrid each channel.
```
Comment (e-koch) [inline docs/smoothing.rst:176]:
```suggestion
<https://reproject.readthedocs.io/en/stable/>`_ package to regrid each channel.
```

File: docs/smoothing.rst [hunk 3]
```diff
@@ -160,3 +160,27 @@ done in the unit of pixel.  In our example, each channel is 0.1 km/s wide::
...
+The regridding is done with interpolation.
+
+It may also be necessary to first spectrally regrid, which can be done
+following the `Spectral Smoothing`_ approach above.
+
+For a full example showing appropriate resampling and reprojection to match two
+cubes, please see the `reprojection tutorial
+<https://github.com/radio-astro-tools/tutorials/pull/17>`_.
```
Comment (e-koch) [inline docs/smoothing.rst:184]: Maybe add a commented-out note that we should change this link once the PR is merged.

**Review by e-koch — CHANGES_REQUESTED** [review]
Minor changes. Then good to go.

**Comment by keflavich** [comment]
superceded by #776 .  Which really just highlights that issue trackers... don't work if you don't read them.

**Pull request closed without merging**